vault/logical_system.go

```diff
@@ -1717,6 +1733,159 @@ func (b *SystemBackend) handleTuneWriteCommon(ctx context.Context, path string,
 		}
 	}
 
+	// user-lockout config
+	{
+		var apiuserLockoutConfig APIUserLockoutConfig
+
+		userLockoutConfigMap := data.Get("user_lockout_config").(map[string]interface{})
+		var err error
+		// Augmenting userLockoutConfigMap for some config options to treat them as comma separated entries
+		err = expandStringValsWithCommas(userLockoutConfigMap)
+		if err != nil {
+			return logical.ErrorResponse(
+					"unable to parse given auth user lockout config information"),
+				logical.ErrInvalidRequest
+		}
+		if userLockoutConfigMap != nil && len(userLockoutConfigMap) != 0 {
+			err := mapstructure.Decode(userLockoutConfigMap, &apiuserLockoutConfig)
+			if err != nil {
+				return logical.ErrorResponse(
+						"unable to convert given user lockout config information"),
+					logical.ErrInvalidRequest
+			}
+
+			switch strings.ToLower(mountEntry.Type) {
+			case "all", "ldap", "approle", "userpass":
+			default:
+				return logical.ErrorResponse("tuning of user lockout configuration for auth type %q not allowed", mountEntry.Type),
+					logical.ErrInvalidRequest
+
+			}
+		}
+
+		if _, ok := userLockoutConfigMap["lockout_threshold"]; ok {
+			var newUserLockoutThreshold int64
+			userLockoutThreshold, err := parseutil.ParseInt(apiuserLockoutConfig.LockoutThreshold)
+			if err != nil {
+				return nil, fmt.Errorf("unable to parse user lockout threshold: %w", err)
+			}
+			oldUserLockoutThreshold := mountEntry.Config.UserLockoutConfig.LockoutThreshold
+
+			switch {
+			case userLockoutThreshold < 0:
+				newUserLockoutThreshold = oldUserLockoutThreshold
```

Comment:
instead of defining a new variable `newUserLockoutThreshold`, would you think we could just `mountEntry.Config.UserLockoutConfig.LockoutThreshold` in the switch in both cases? Might be more readable.